clang/lib/Driver/ToolChains/BareMetal.cpp

```diff
@@ -296,6 +300,47 @@ void BareMetal::AddClangCXXStdlibIncludeArgs(const ArgList &DriverArgs,
     return;
 
   const Driver &D = getDriver();
+  std::string Target = getTripleString();
+
+  auto AddCXXIncludePath = [&](StringRef Path) {
```

Comment:
Is `Target` the only thing you're capturing here? Perhaps just pass it in at the lambda invocation site?

Reply:
No, we also need `this` for `detectLibcxxVersion`, `DriverArgs` and `CC1Args`.